agent/dns.go

```diff
@@ -243,6 +245,8 @@ func (d *DNSServer) ListenAndServe(network, addr string, notif func()) error {
 	if network == "udp" {
 		d.UDPSize = 65535
 	}
+
+	d.MaxTCPQueries = cfg.MaxTCPQueries
```

Comment:
This might need to be set on the dns.Server record, e.g. d.Server.MaxTCPQueries to have the effect you want.

Reply:
I think this is correct because `Server` is embedded in `DNSServer` so `d.MaxTCPQueries` is equivalent to `d.Server.MaxTCPQueries`